src/components/CodeEditor/CodeEditor.js

```diff
@@ -265,12 +286,21 @@ class CodeEditor extends Component {
     this._mountNode = ref;
   };
 
-  _updateState(code) {
+  _updateState(code, showJSX = true) {
     try {
-      return {
-        compiled: compile(code),
+      let newState = {
+        compiled: compileES5(code),
```

Comment:
Should `code` always go into `newState`? What's the benefit of only adding it conditionally.

Reply:
I think no because we want to keep code only if we are in JSX view. 

If we mix code from ES5 view to ES6 it wont work any more.